src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,70 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
+import { FilterBy } from '../utiles/FilterBy';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+interface Props {
+  qerty: string;
+  setQerty: (qerty: string) => void;
+  filterBy: FilterBy;
+  setFilterBy: (filterBy: FilterBy) => void;
+}
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  qerty,
+  setQerty,
+  filterBy,
+  setFilterBy,
+}) => {
+  const handleImputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQerty(event.target.value);
+  };
+
+  const handleFilter = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setFilterBy(event.target.value as FilterBy);
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={filterBy}
+            onChange={handleFilter}
+          >
+            <option value={FilterBy.All}>All</option>
+            <option value={FilterBy.Active}>Active</option>
+            <option value={FilterBy.Completed}>Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          name="searchInput"
+          value={qerty}
+          onChange={handleImputChange}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+          {qerty && (
+          // eslint-disable-next-line jsx-a11y/control-has-associated-label
+            <button
+              data-cy="clearSearchButton"
+              type="button"
+              className="delete"
+              aria-label="clear"
+              onClick={() => setQerty('')}
+            />
```

Comment:
```suggestion
            <button
              data-cy="clearSearchButton"
              type="button"
              className="delete"
              aria-label="clear"
              onClick={() => setQerty('')}
            />
```
as you set aria-label - disable eslint can be removed